File: basaltpass-backend/internal/model/invitation.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// Invitation 团队邀请
// status: pending / accepted / rejected / revoked

type Invitation struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	TeamID    uint           `gorm:"index;not null" json:"team_id"`
	Team      Team           `json:"team"`
	InviterID uint           `gorm:"index;not null" json:"inviter_id"`
	Inviter   User           `json:"inviter"`
	InviteeID uint           `gorm:"index;not null" json:"invitee_id"`
	Invitee   User           `json:"invitee"`
	Status    string         `gorm:"size:16;not null;default:'pending'" json:"status"`
	Remark    string         `gorm:"size:255" json:"remark"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

File: basaltpass-frontend/src/shared/api/user/invitation.ts
import client from '../client'

export interface Invitation {
  id: number
  team_id: number
  inviter_id: number
  invitee_id: number
  status: 'pending' | 'accepted' | 'rejected' | 'revoked'
  remark?: string
  created_at: string
  team?: { id: number; name: string }
  inviter?: { id: number; nickname: string; email: string }
  invitee?: { id: number; nickname: string; email: string }
}

export const invitationApi = {
  // translatedinvitation
  create: (teamId: number, inviteeIds: number[], remark = '') =>
    client.post(`/api/v1/teams/${teamId}/invitations`, { invitee_ids: inviteeIds, remark }),

  // translatedinvitation
  revoke: (teamId: number, invId: number) =>
    client.delete(`/api/v1/teams/${teamId}/invitations/${invId}`),

  // translatedinvitation
  accept: (invId: number) => client.put(`/api/v1/invitations/${invId}/accept`),

  // translatedinvitation
  reject: (invId: number) => client.put(`/api/v1/invitations/${invId}/reject`),

  // translatedtoinvitation
  listIncoming: () => client.get<Invitation[]>('/api/v1/invitations'),

  // teamalreadytranslatedinvitation
  listOutgoing: (teamId: number) => client.get<Invitation[]>(`/api/v1/teams/${teamId}/invitations`),
} 